src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/perform_interaction_request.cc

```diff
@@ -1127,43 +1117,58 @@ mobile_apis::Result::eType
 PerformInteractionRequest::PrepareResultCodeForResponse(
     const app_mngr::commands::ResponseInfo& ui_response,
     const app_mngr::commands::ResponseInfo& vr_response) {
-  LOG4CXX_DEBUG(logger_,
-                "InteractionMode = " << static_cast<int32_t>(interaction_mode_)
-                                     << " | FirstAnsweredInterface = "
-                                     << static_cast<int32_t>(first_responder_));
-
-  if (mobile_apis::InteractionMode::VR_ONLY == interaction_mode_) {
-    if (FirstAnsweredInterface::VR == first_responder_) {
-      return MessageHelper::HMIToMobileResult(vr_result_code_);
+  LOG4CXX_DEBUG(
+      logger_, "InteractionMode = " << static_cast<int32_t>(interaction_mode_));
+
+  auto result = mobile_api::Result::INVALID_ENUM;
+  switch (interaction_mode_) {
+    case mobile_apis::InteractionMode::VR_ONLY: {
+      result = MessageHelper::HMIToMobileResult(vr_result_code_);
+      break;
     }
-  }
 
-  if (mobile_apis::InteractionMode::BOTH == interaction_mode_) {
-    if (IsVRPerformInteractionResponseSuccessfulInBothMode()) {
-      return MessageHelper::HMIToMobileResult(vr_result_code_);
+    case mobile_apis::InteractionMode::MANUAL_ONLY: {
+      result = MessageHelper::HMIToMobileResult(ui_result_code_);
+      break;
+    }
+
+    case mobile_apis::InteractionMode::BOTH: {
+      if (IsVRPerformInteractionResponseSuccessfulInBothMode()) {
+        return MessageHelper::HMIToMobileResult(vr_result_code_);
+      } else {
+        return MessageHelper::HMIToMobileResult(ui_result_code_);
+      }
+      break;
     }
-    return MessageHelper::HMIToMobileResult(ui_result_code_);
+
+    default:
+      result = CommandRequestImpl::PrepareResultCodeForResponse(ui_response,
+                                                                vr_response);
   }
 
-  return CommandRequestImpl::PrepareResultCodeForResponse(ui_response,
-                                                          vr_response);
+  return result;
 }
 
 bool PerformInteractionRequest::PrepareResultForMobileResponse(
     app_mngr::commands::ResponseInfo& ui_response,
     app_mngr::commands::ResponseInfo& vr_response) const {
-  if (mobile_apis::InteractionMode::VR_ONLY == interaction_mode_) {
-    if (FirstAnsweredInterface::VR == first_responder_) {
-      return vr_response.is_ok;
-    }
-  }
-
-  if (mobile_apis::InteractionMode::BOTH == interaction_mode_) {
-    return (vr_response.is_ok || ui_response.is_ok);
+  bool result = false;
```

Comment:
again i think a switch is too verbose, especially since there is no default case